services/src/main/scala/cromwell/services/instrumentation/CromwellInstrumentation.scala

```diff
@@ -0,0 +1,106 @@
+package cromwell.services.instrumentation
+
+import akka.actor.{Actor, ActorRef, Cancellable}
+import cats.data.NonEmptyList
+import com.typesafe.config.ConfigFactory
+import cromwell.services.instrumentation.CromwellInstrumentation.{InstrumentationPath, _}
+import cromwell.services.instrumentation.InstrumentationService.InstrumentationServiceMessage
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration._
+import scala.language.implicitConversions
+
+object CromwellInstrumentation {
+  
+  val InstrumentationRate = ConfigFactory.load()
+    .getConfig("system")
+    .as[Option[FiniteDuration]]("instrumentation-rate")
+    .getOrElse(5.seconds)
+
+  type InstrumentationPath = NonEmptyList[String]
+  
+  implicit def stringToNel(str: String): NonEmptyList[String] = NonEmptyList.of(str)
+  
+  implicit class EnhancedStatsDPath(val path: InstrumentationPath) extends AnyVal {
+    def withStatusCodeFailure(code: Option[Int]) = code
+      .map(c => path.concat(NonEmptyList.of(c.toString)))
+      .getOrElse(path)
+
+    def withThrowable(failure: Throwable, statusCodeExtractor: Throwable => Option[Int]) = {
+      path.withStatusCodeFailure(statusCodeExtractor(failure))
+    }
+  }
+}
+
+trait CromwellInstrumentation {
```

Comment:
ToL This this feels coupled to the actor model.  I was actually expecting to fire off UDP right away, as I thought batching was being done at the StatsD "collector."

That being said 
 * I don't want to hold this up
 * We could refactor if we really wanted to do it differently

Reply:
Yep it is coupled to the actor model on purpose. By putting it behind the service registry we can easily swap out the current implementation for another one, even for something else than StatsD. With the current implementation there's no real batching. It re-uses what Rob did for workbench, explained here https://broadinstitute.atlassian.net/wiki/spaces/GAWB/pages/118259719/Workbench+Metrics

It's a tradeoff between tying ourselves to a specific implementation or to the actor model and IMO since Cromwell is full of actors anyway I'd rather it being tied to the actor model and decouple the underlying implementation.